From fe3486178040db782f18d67a95a1e63bdf68eedf Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Sat, 1 May 2021 16:30:59 -0400
Subject: [PATCH] sim: nrun: add local strsignal prototype

While libiberty provides a definition for this for systems that lack
the function (e.g. Windows), it doesn't provide a prototype.  So add
our own local copy in the one file that uses the func.

diff --git a/sim/common/nrun.c b/sim/common/nrun.c
index 0caa82062a2..1ceb0837e2c 100644
--- a/sim/common/nrun.c
+++ b/sim/common/nrun.c
@@ -28,6 +28,11 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.  */
 
 #include "bfd.h"
 
+#ifndef HAVE_STRSIGNAL
+/* While libiberty provides a fallback, it doesn't provide a prototype.  */
+extern const char *strsignal (int);
+#endif
+
 #ifdef HAVE_ENVIRON
 extern char **environ;
 #endif
